lib/index.js

```diff
@@ -349,7 +373,8 @@ exports.IncompatibleSchemas = typedErrors.makeTypedError('IncompatibleSchemas')
  */
 exports.match = (() => {
 	const ajv = new AJV({
-		allErrors: true
+		allErrors: true,
+		schemaId: '$id'
```

Comment:
What do we need this option for?

Reply:
Its to enfore that `$id` is the only valid identifier property (this is important as schemas can reference each other by identifier), its a very minor thing but I dislike that both `id` and `$id` may be present. This option will ignore `id` and also log a warning if present

Reply:
Actually, this means we should remove the `|| schema.id` when grabbing the identifier